fdbserver/ClusterController.actor.cpp

```diff
@@ -2434,8 +2434,7 @@ ACTOR Future<Void> startEncryptKeyProxy(ClusterControllerData* self, double wait
 			// hence, the process only waits for the master recruitment and not the full cluster recovery.
 			state bool noEncryptKeyServer = !self->db.serverInfo->get().encryptKeyProxy.present();
 			while (!self->masterProcessId.present() ||
-			       self->masterProcessId != self->db.serverInfo->get().master.locality.processId() ||
-			       self->db.serverInfo->get().recoveryState < RecoveryState::LOCKING_CSTATE) {
+			       self->masterProcessId != self->db.serverInfo->get().master.locality.processId()) {
```

Comment:
@sfc-gh-ahusain I suspect the waiting for `>= locking_cstate` is because we want to wait for the the master process to be more "sticky". Like, if the cluster recovery is able to go pass the reading_cstate state, the master is more likely to finish the cluster recovery.

I have run 100K correctness to verified correctness is happy without the recovery state wait. If we couldn't find out the reason for having the logic, I would say let's remove it.

Reply:
@sfc-gh-yiwu, sounds like a good plan.

Reply:
This code is because there is an interaction effect between the singleton recruitment and the transaction sub-system recruitment. If I have 5 stateless process and want to configure 3 commit proxies, then ideally we would recruit those first and then let the other singletons share the remaining stateless processes.

Removing this with not cause correctness issues (as you have shown) but it will cause multiple recoveries in a row to settle on the correct configuration in some cases, which would be bad in real world cases.